core/common/src/ITwinError.ts

```diff
@@ -0,0 +1,92 @@
+import { BentleyError, LoggingMetaData } from "@itwin/core-bentley";
+import { LockState } from "./IModelError";
+
+/**
+ * Detailed information about a particular object Lock that is causing the Lock update conflict.
+ * An example of a lock update conflict would be attempting to use [LockControl.acquireLocks]($backend) on an object that is already locked by another Briefcase.
+ * @beta
+ */
+export interface InUseLock {
+/** Id of the object that is causing conflict. */
+  objectId: string;
+  /**
+   * The level of conflicting lock. Possible values are {@link LockState.Shared}, {@link LockState.Exclusive}.
+   * See {@link LockState}.
+   */
+  state: LockState;
+  /** An array of Briefcase ids that hold this lock. */
+  briefcaseIds: number[];
+}
+
+/**
+ * An interface used to describe an error. This error interface should be extended when needing to throw errors with extra properties defined on them. See [[InUseLocksError]] for an example.
```

Comment:
Please make it much clearer that it describes an error **for a developer/application**. The message is not intended to be displayed to an end user. Be more explicit about how namespace is a uniqueness qualifier for key, which must itself be unique within its namespace. Ideally, provide an example code snippet where some function receives an `ITwinError`, determines that it is an `InUseLocksError`, and accesses the `inUseLocks` property of that derived type.

Reply:
I added the example code snippet, and added extra text to the message, namespace and errorkey properties capturing that information.

Reply:
Not to nitpick but example code snippets should go in example-code/snippets and be referenced by docs, to ensure they compile now and in the future.

Reply:
I gave it a try. Let me know if it looks off.